### PR TITLE
Add support for zstd compression & Fix Automatic Decoding

### PR DESCRIPTION
**Pull request by shrimpwtf**
The changes in this commit include adding support for zstd compression in transport.go. This was achieved by importing the "compress/zstd" package and creating a new 'zstdReader' type. Also, the zstdReader was integrated into the existing system by handling zstd in the switch-case statement for identifying the compression method.

I've also included a patch to fix automatic gzip encoding, as it only currently works if you explicitly do NOT set a accept-encoding header, if you set one it expects you to do it yourself. This helps mimic chrome fingerprints more closely and fix weird garbled responses like the below.

Please test and edit as need be.

CleanShot 2024-06-11 at 16 50 54@2x

### CODE REVIEW

File: transport.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ import (
 	"compress/flate"
 	"compress/gzip"
 	"compress/zlib"
+	"compress/zstd"
```
Comment (Dissociable) [inline transport.go:18]: This is not a valid package as far as I can see.
Replacing it with `github.com/klauspost/compress/zstd` is fine and compatible.

File: transport.go [hunk 2]
```diff
@@ -2567,6 +2568,11 @@ func (pc *persistConn) roundTrip(req *transportRequest) (resp *Response, err err
 		req.extraHeaders().Set("Accept-Encoding", "gzip, deflate, br")
 	}
 
+	// Set gzip to true if the caller's Accept-Encoding includes gzip.
+	if strings.Contains(req.Header.Get("Accept-Encoding"), "gzip") || strings.Contains(req.Header.get("accept-encoding"), "gzip") {
```
Comment (Dissociable) [inline transport.go:2572]: As the comments of `Header.Get` states, key is case-insensitive as `textproto.CanonicalMIMEHeaderKey` is used to canonicalize the provided key.
So the first check is enough.

**Review by Dissociable — CHANGES_REQUESTED** [review]

**Comment by ll0utr3** [comment]
bump, it would allow to match chrome and firefox 'Accept-Encoding' header

File: transport.go (hunk 1 above)
Comment (bogdanfinn) [reply transport.go:18]: @Dissociable thank you

**Comment by bogdanfinn** [comment]
@shrimpwtf @Dissociable thank you very much both for your contribution

**Pull request merged**